ligo/haskell/test/Test/Ligo/BaseDAO/Proposal.hs

```diff
@@ -9,62 +9,167 @@ import Universum
-          withSender (AddressResolved owner1) $ getTotalSupply unfrozenTokenId consumer
+          withSender (AddressResolved proposer) $ getTotalSupply unfrozenTokenId consumer
           checkStorage (AddressResolved $ toAddress consumer) (toVal [148 :: Natural]) -- initial = 200
 
           consumer2 <- originateSimple "consumer" [] contractConsumer
-          withSender (AddressResolved owner1) $ getTotalSupply frozenTokenId consumer2
+          withSender (AddressResolved proposer) $ getTotalSupply frozenTokenId consumer2
           checkStorage (AddressResolved $ toAddress consumer2) (toVal [52 :: Natural]) -- initial = 0
 
     , nettestScenario "cannot propose with insufficient tokens to pay the fee" $
         uncapsNettest $ do
-          ((owner1, _), _, dao, admin) <- originateLigoDao
+          ((proposer, _), _, dao, admin) <- originateLigoDao
           withSender (AddressResolved admin) $
-            call dao (Call @"Set_fixed_fee_in_token") (100 :: Natural)
+            call dao (Call @"Set_fixed_fee_in_token") 100
 
           let params = ProposeParams
                 { ppFrozenToken = 1
                 , ppProposalMetadata = proposalMetadataFromNum 1
                 }
-          withSender (AddressResolved owner1) $ call dao (Call @"Propose") params
+          withSender (AddressResolved proposer) $ call dao (Call @"Propose") params
             & expectCustomErrorNoArg #pROPOSAL_INSUFFICIENT_BALANCE
+
+    , nettestScenario "an owner can change the fee" $
+        uncapsNettest $ do
+          ((proposer, _), _, dao, admin) <- originateLigoDao
+          withSender (AddressResolved admin) $
+            call dao (Call @"Set_fixed_fee_in_token") 1000
+
+          let params = ProposeParams
+                { ppFrozenToken = 0
+                , ppProposalMetadata = proposalMetadataFromNum 1
+                }
+          withSender (AddressResolved proposer) $ call dao (Call @"Propose") params
+            & expectCustomErrorNoArg #pROPOSAL_INSUFFICIENT_BALANCE
+
+          withSender (AddressResolved admin) $
+            call dao (Call @"Set_fixed_fee_in_token") 10
+
+          withSender (AddressResolved proposer) $ call dao (Call @"Propose") params
+          checkTokenBalance frozenTokenId dao proposer 10
+          checkTokenBalance unfrozenTokenId dao proposer 90
+
+    , nettestScenario "a non-owner cannot change the fee" $
+        uncapsNettest $ do
+          ((someone, _), _, dao, _) <- originateLigoDao
+
+          withSender (AddressResolved someone) $
+            call dao (Call @"Set_fixed_fee_in_token") 1000
+              & expectCustomErrorNoArg #nOT_ADMIN
+
+    , nettestScenario "a proposer is returned a fee after the proposal succeeds" $
+        uncapsNettest $ do
+          ((proposer, _), (voter, _), dao, admin) <- originateLigoDao
+
+          -- Use 60s for voting period, since in real network by the time we call
+          -- vote entrypoint 30s is already passed.
```

Comment:
```suggestion
          -- the vote entrypoint 30s have already passed.
```
Nitpick as well.

Leaving the comment was a good idea tho, there's also other tests that are time-dependent and we'll need to keep this in mind.

Reply:
I found the code with `advanceTime` in another module, and copied it with this comment (it's indeed helpful). I think I should fix the grammar there as well.

Reply:
Oh I didn't know that.

Yes, please fix that as well :ok_hand: